lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -627,6 +596,83 @@ function writeStringsToCtcFiles(locale, strings) {
   fs.writeFileSync(fullPath, JSON.stringify(output, null, 2) + '\n');
 }
 
+/**
+ * @param {Record<string, CtcMessage>} strings
+ */
+function checkForCollisions(strings) {
```

Comment:
this code has a few purposes:

- mutate ctc messages that collide to add `meaning` set to `description`
- throw if the meaning of any collisions also collides
- throw if the number of collisions has changed

I think `checkForCollisions` captures the second two but doesn't capture the first so well. It's also kinda not doing a great job at the second bullet either. Maybe we should rewrite this altogether?

Proposal:
- group by message
- set meaning on all groups of length > 1
- assert the equality of all groups of length > 1 with previous run
- throw if the `meaning` set size is less than the group length for any given group

I know that's more than you probably bargained for though, so I'm happy to take it and switch roles on this PR if you prefer :)

Reply:
ha, I was sniped by this to do some fancy stuff on top of the base fix in #12698, so we can have our pick of approaches...

Reply:
> I know that's more than you probably bargained for though

Yeah... I'd prefer to defer to you or brendan for all that. Just keep the `expect` thing and LGTM :)

Reply:
done :)